arrows/klv/klv_read_write.txx

```diff
@@ -0,0 +1,378 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// Implementation of the templated basic KLV read/write functions.
+
+#include "klv_read_write.h"
+
+#include <vital/exceptions.h>
+
+#include <limits>
+#include <numeric>
+#include <string>
+
+#include <cmath>
+
+namespace kwiver {
+  namespace arrows {
+    namespace klv {
+// ---------------------------------------------------------------------------
+// Macros for verifying type attributes
+// Message-less static_assert not available in C++11
+
+#define KLV_ASSERT_INT( T ) \
+  static_assert( std::is_integral < T > ::value, "must be an integer type" )
+
+#define KLV_ASSERT_UINT( T )                                                \
+  KLV_ASSERT_INT( T );                                                      \
+  static_assert( std::is_unsigned < T > ::value, "must be an unsigned type" )
+
+#define KLV_ASSERT_SINT( T )                                           \
+  KLV_ASSERT_INT( T );                                                 \
+  static_assert( std::is_signed < T > ::value, "must be a signed type" )
+
+#define KLV_ASSERT_UINT8_ITERATOR( ITER )                              \
+  static_assert(                                                       \
+    std::is_same < typename std::decay < decltype( *ITER ) > ::type,      \
+    uint8_t > ::value, "iterator must point to uint8_t" )
+
+// ---------------------------------------------------------------------------
+// Returns number of bits required to store the given signed or unsigned int.
```

Comment:
I'm not saying this is wrong, but the full indents before code is surprising to me._